ADAL/src/request/ADAuthenticationRequest.m

```diff
@@ -85,6 +89,16 @@ + (ADAuthenticationRequest*)requestWithContext:(ADAuthenticationContext*)context
     return request;
 }
 
+- (id)init
+{
+    if (!(self = [super init]))
+        return nil;
+    
+    sInteractionInProgress = dispatch_semaphore_create(1);
```

Comment:
This is not the right place to initialize a static variable. You'll end up recreating the semaphore every time a new ADAuthenticationRequest is created.

Put this in + (void)initialize